coverart.lua

```diff
@@ -255,9 +258,11 @@ local function addFromDirectory(directory, is_icy)
 
     --loops through the all the files in the directory to find if any are valid cover art
     local success = false
+    local first_ok
     for i, file in ipairs(files) do
         --if the name matches one in the whitelist then load it
-        if isValidCoverart(file, is_icy) then
+        local valid = isValidCoverart(file, is_icy)
```

Comment:
May want to consider changing the variable name `valid` as well. It's not a boolean anymore.